src/pages/docs/chat/rooms/media-sharing.mdx

```diff
@@ -0,0 +1,165 @@
+
+```javascript
+function createMessageDOM(message) {
+  const container = document.createElement("div");
+  const text = document.createElement("div");
+  text.innerText = message.text;
+  container.appendChild(text);
+  const validImages = getValidImages(message);
+  if (validImages.length > 0) {
+    const imagesContainer = document.createElement("div");
+    for (let imageId of validImages) {
+      const img = document.createElement("img");
+      img.src = "https://example.com/images/"+imageId;
+      imagesContainer.appendChild(img);
+    }
+    container.appendChild(imagesContainer);
+  }
+  return container;
+}
+```
+
+### Why image ID and not URL?
+
+Image IDs are simpler to validate than full URLs. You can also use URLs, but you must correctly validate them on the receiving end.
+
+### Metadata for each image
+
+In the example above we used a simple string array to store image IDs. We can change this to be an object where each image holds extra information that can be displayed to the user, such as a title or description, or the expected size in pixels so that the UI can allocate a placeholder loading box of the correct proportions.
+
+### Add an image to an existing message
+
+You can add an image to an existing message by editing its metadata.
+
+```javascript
+let imageId = 'abcd123abcd'; // assume this is the image we want to add
+let message = ...; // assume this is the message we want to edit
+
+const newMetadata = structuredClone(message.metadata);
+if (!newMetadata.images) {
+  newMetadata.images = [];
+}
+newMetadata.images.push(imageId);
+
+room.messages.update(message.serial, message.copy({metadata: newMetadata}))
+```
+
+### Remove an image from an existing message
+
+Similarly to how we add a new image, we can remove existing images from messages by updating the message:
+
+```javascript
+let imageId = 'abcd123abcd'; // assume this is the image we want to remove
+let message = ...; // assume this is the message we want to edit
+
+if (!message.metadata.images || message.metadata.images.length === 0) {
+  //do nothing if there are no images
+  return; 
+}
+const newMetadata = structuredClone(message.metadata);
+newMetadata.images = newMetadata.images.filter(id => imageId !== id);
+
+room.messages.update(message.serial, message.copy({metadata: newMetadata}))
+```
+
+## Privacy and security considerations
+
+The message metadata is not validated by the server. Always treat it as untrusted user input. The same applies to the text and headers of a chat message.
+
+### Validate metadata on the receiving end
+
+You must always validate the metadata before displaying any media to the user. Treat the metadata as untrusted user input. If you are using IDs, validate they are of the correct format. If you are using URLs, validate the schema, domain, path and query parameters are what you expect. Do not display images or other media directly from the metadata of a message.
+
+### Use authentication server-side to control access
```

Comment:
Should this be higher up the page? Seems important to understand the access controls before setting it up.